src/azure-cli/azure/cli/command_modules/synapse/manual/_client_factory.py

```diff
@@ -105,17 +105,16 @@ def cf_kusto_pool(cli_ctx, *_):
 def synapse_spark_factory(cli_ctx, workspace_name, sparkpool_name):
     from azure.synapse.spark import SparkClient
     from azure.cli.core._profile import Profile
+    from azure.cli.core.auth.util import resource_to_scopes
     from azure.cli.core.commands.client_factory import get_subscription_id
     subscription_id = get_subscription_id(cli_ctx)
     profile = Profile(cli_ctx=cli_ctx)
-    cred, _, _ = profile.get_login_credentials(
-        resource=cli_ctx.cloud.endpoints.synapse_analytics_resource_id,
-        subscription_id=subscription_id
-    )
+    cred, _, _ = profile.get_login_credentials(subscription_id=subscription_id)
     return SparkClient(
         credential=cred,
         endpoint='{}{}{}'.format("https://", workspace_name, cli_ctx.cloud.suffixes.synapse_analytics_endpoint),
-        spark_pool_name=sparkpool_name
+        spark_pool_name=sparkpool_name,
+        credential_scopes=resource_to_scopes(cli_ctx.cloud.endpoints.synapse_analytics_resource_id)
```

Comment:
`SparkClient`'s handling of `credential_scopes` is wrong.

In `azure.synapse.spark._configuration.SparkClientConfiguration.__init__`

```py
        self.credential_scopes = ['https://dev.azuresynapse.net/.default']
        self.credential_scopes.extend(kwargs.pop('credential_scopes', []))
```

`credential_scopes` is added to the existing `['https://dev.azuresynapse.net/.default']`, instead of replacing it. See https://github.com/Azure/autorest.python/pull/745.

https://github.com/Azure/azure-cli/pull/29690 dropped support for this incorrect usage.